test/perf/test_performance.py

```diff
@@ -0,0 +1,86 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+import logging
+import pytest
+from time import time
+from tempeh.configurations import models, datasets
+
+from fairlearn.post_processing import ThresholdOptimizer
+from fairlearn.post_processing.threshold_optimizer import DEMOGRAPHIC_PARITY
+from fairlearn.reductions import ExponentiatedGradient, GridSearch
+from fairlearn.reductions.moments import DemographicParity
+
+from conftest import get_all_perf_test_configurations
+
+all_perf_test_configurations = get_all_perf_test_configurations()
+all_perf_test_configurations_descriptions = \
+    [config.__repr__().replace(' ', '') for config in all_perf_test_configurations]
+
+
+logging.basicConfig(level=logging.DEBUG)
+
+
+@pytest.mark.parametrize("perf_test_configuration", all_perf_test_configurations,
+                         ids=all_perf_test_configurations_descriptions)
+def test_perf(perf_test_configuration):
+    print("Downloading dataset")
+    dataset = datasets[perf_test_configuration.dataset]()
+    print("Done downloading dataset")
+
+    if perf_test_configuration.dataset == "adult_uci":
+        # sensitive feature is 8th column (sex)
+        sensitive_features_train = dataset.X_train[:, 7]
+        sensitive_features_test = dataset.X_test[:, 7]
+    elif perf_test_configuration.dataset == "diabetes_sklearn":
+        # sensitive feature is 2nd column (sex)
+        # features have been scaled, but sensitive feature needs to be str or int
+        sensitive_features_train = dataset.X_train[:, 1].astype(str)
+        sensitive_features_test = dataset.X_test[:, 1].astype(str)
+        # labels can't be floats as of now
+        dataset.y_train = dataset.y_train.astype(int)
+        dataset.y_test = dataset.y_test.astype(int)
+    elif perf_test_configuration.dataset == "compas":
+        # sensitive feature is either race or sex
+        # TODO add another case where we use sex as well, or both (?)
+        sensitive_features_train = dataset.race_train
+        sensitive_features_test = dataset.race_test
+    else:
+        raise ValueError("Sensitive features unknown for dataset {}"
+                         .format(perf_test_configuration.dataset))
+
+    print("Fitting estimator")
+    estimator = models[perf_test_configuration.predictor]()
```

Comment:
Might be interesting to know how long the 'raw' estimator takes.... are nested timers available?

Reply:
I mean right now it's all fairly "dumb" in the sense that we just count from start to end. We can think about creating a way to expose that. Don't want to do it in this PR, though.